agent-inject/agent/container_env.go

```diff
@@ -10,6 +10,11 @@ import (
 func (a *Agent) ContainerEnvVars(init bool) ([]corev1.EnvVar, error) {
 	var envs []corev1.EnvVar
 
+	envs = append(envs, corev1.EnvVar{
+		Name:  "VAULT_TOKEN",
+		Value: "/home/vault/.vault-token",
```

Comment:
Shouldn't this environment variable hold the contents of the token instead? cf. https://www.vaultproject.io/docs/commands/#vault_token

Reply:
There's a bug with Kube auth login and namespaces that we haven't been able to track down, so setting this was the only way to get it to work. Auto-auth populates file cache which takes priority. This could be any value.

Reply:
I can put a comment in about this being a work around until we find the bug in the auth method.

Reply:
I see. I think this might affect the auto-revoke on shutdown that I implemented in #67

Reply:
I will test the two together and let you know.

Reply:
@lawliet89 I found a less intrusive solution.  This shouldn't cause issues for you!